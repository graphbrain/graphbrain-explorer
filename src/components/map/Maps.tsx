
import React, { useState, useEffect } from 'react';

import ForcedMap from './ForcedMap';
import Loader from '../Loader';
import Error from '../Error'

import backIcon from '../../assets/back_icon.svg';

import {getData} from '../../api/getData';

import '../../styles/map.scss';

import { Topic } from '../LandingPage';

export interface Node {
  faction: number,
  id: string,
  index: 0,
  info: {gender_numner: string},
  label: string,
  vx: number,
  vy: number,
  weight: 2,
  x: number,
  y: number
}

export interface Link {
  id: string,
  directed: boolean,
  index: number,
  info: {
    headlines: Array<string>,
    other_topics: Array<Topic>
  },
  label: string,
  source: Node,
  target: Node,
  type: string,
  weight: number
}

export interface Data {
  layout: string,
  links: Array<Link>,
  nodes: Array<Node>,
  topic_label: string,
  type: string
}

const Maps: React.FC<{}> = () => {
   
  const [data, setData] = useState<Data | null>(null);
  const [error, setError] = useState<boolean>(false)

  useEffect(() => {
    getData(window.location.search).then(data => {
      if (data && data["viz_blocks"]) {
        setData(data["viz_blocks"][0]);
      }
      else setError(true)
  });
  }, [])

  const handleBackClick = () => {
    window.location.href = '/';
  }

  return (
    <div className="mapsArea">
    {data && (
      <div className="mapHeader">
        <img src={backIcon} alt="back" className="backIcon" aria-label="backIcon" onClick={handleBackClick}/>
        <h2 className="mapTitle">
            {data && data.topic_label && data.topic_label.length === 2 ? 
              data.topic_label.toUpperCase() : 
              data.topic_label}
        </h2> 
        <ForcedMap data={data}/>
      </div>
      )}
      {!data && !error && <Loader />}
      {error && <Error/>}
    </div>
  )
}

export default Maps;