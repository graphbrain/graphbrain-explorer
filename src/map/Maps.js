
import React, { useState, useEffect } from 'react';

import ForcedMap from './ForcedMap';

import Loader from '../components/Loader';

import backIcon from '../assets/back_icon.svg';

import {getData} from '../api/getData';

import './map.scss';


const Maps = () => {
   
  const [data, setData] = useState(null);

  useEffect(() => {
    getData(window.location.search).then(data => {
     setData(data["viz_blocks"][0]);
  });
  }, [])

  const handleBackClick = () => {
    window.location.href = '/';
  }

  const mapToRender = () => {
    if (data) {
      return <ForcedMap data={data}/>;
    } 
    else return <Loader />;  
  }

  return (
    <div className="mapsArea">
    {data && (
      <div className="mapHeader">
        <img src={backIcon} alt="back" className="backIcon" onClick={handleBackClick}/>
        <h2 className="mapTitle">
            {data.topic_label.length === 2 ? 
              data.topic_label.toUpperCase() : 
              data.topic_label}
        </h2> 
      </div>
      )}
      {mapToRender()}
    </div>
  )
}

export default Maps;