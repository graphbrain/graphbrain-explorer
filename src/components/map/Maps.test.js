import React from "react";
import { render, fireEvent } from "@testing-library/react";

import '@testing-library/jest-dom/extend-expect';

import Maps from "./Maps";


describe("<Maps />", () => {

  const data = {
    links: [{id: "1"}, {id: "2"}],
    nodes:  [{id: "1", faction: 1}, {id: "2", faction: 1}],
    topic_label: "yemen"
  }

  beforeEach(() => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve({ data: data }),
      })
    );
  });

  it("renders correctly", () => {   
    const { asFragment } = render(<Maps />);
    expect(asFragment()).toMatchSnapshot();   
  });

  it('renders a loader while fetching data', () => {
    const { getByTestId } = render(<Maps />);
    expect(getByTestId('loader')).toBeTruthy();
  })

  it ("gets you back to landing page when clicking back", async () => {
    setTimeout(() => {
      const { container, getByTestId } = render(<Maps/>);
      const backButton  = getByTestId('backIcon');
      fireEvent.click(backButton);
      expect(container.querySelector('.mapsArea')).toBeNull();
    }, 500)
   
  })

  it('renders the map once data is fetched', async () => {
    const { getByTestId } = render(<Maps />);
    setTimeout(() => {
      expect(getByTestId('contentArea')).toBeTruthy();
    }, 500)
  })

});

