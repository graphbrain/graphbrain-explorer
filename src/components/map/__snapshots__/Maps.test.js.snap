// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Maps /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mapsArea"
  >
    <div
      class="loader"
    />
  </div>
</DocumentFragment>
`;
